### PR TITLE
Enhancement

### PR DESCRIPTION
**Pull request by anantjain6**
1. Partially Fixed tab spacing in EventController
2. At the time of teacher login array of events ID created by him are put to session and from it middleware EventOwner will check that whether the event he is trying to access belongs to him
3. Added on-delete to foreign key migration
4. Added favicon and chrome address bar colour
5. Partially completed work of displaying event info when a person is not login
6. Displayed number of question for every event in a student panel

### CODE REVIEW

File: app/Http/Controllers/EventController.php [hunk 1]
```diff
@@ -15,103 +15,117 @@ class EventController extends Controller
...
+	    {
+	    	//Add new Subject to subject table
+	    	$subject = new Subject;
+	    	$subject->name = $request->newsubject;
+	    	$subject->save();
+	    	$task->subid = $subject->id;
+	    }
+	    else
+	    {
+	    	Subject::findOrFail($request->subject);
```
Comment (ankitjain28may) [inline app/Http/Controllers/EventController.php]: Are you handling the error ?

Comment (ankitjain28may) [reply app/Http/Controllers/EventController.php]: findOrFail will throw Model Not Found exception so, are you showing any message to the user?

File: app/Http/Controllers/EventController.php [hunk 2]
```diff
@@ -15,103 +15,117 @@ class EventController extends Controller
...
-			        'correct_mark' => 'required|numeric|not_in:0',
-			        'wrong_mark' => 'required|numeric',
-			        'display_ques' => 'required|numeric|not_in:0',
-			        'newsubject' => 'sometimes|not_in:subject',
-			    ]
-				,[
-					'not_in' => 'The :attribute field is required.'
-				]);
+	    $this -> validate($request,
+		    	[
```
Comment (ankitjain28may) [inline app/Http/Controllers/EventController.php]: Some are still tabs

**Comment by anantjain6** [comment]
Commit eefea1e added

File: app/Exceptions/Handler.php [hunk 3]
```diff
@@ -44,7 +44,8 @@ public function report(Exception $exception)
      */
     public function render($request, Exception $exception)
     {
-        return parent::render($request, $exception);
+        return view('errors.404');
```
Comment (ankitjain28may) [inline app/Exceptions/Handler.php]: It's good that you did this..

File: app/Http/Controllers/EventController.php [hunk 4]
```diff
@@ -15,222 +15,224 @@ class EventController extends Controller
...
+        {
+            //Add new Subject to subject table
+            $subject = new Subject;
+            $subject->name = $request->newsubject;
+            $subject->save();
+            $task->subid = $subject->id;
+        }
+        else
+        {
+            $sub = Subject::select('id')->where('id',$request->subject)->first();
```
Comment (ankitjain28may) [inline app/Http/Controllers/EventController.php]: Better

File: app/Http/Controllers/EventController.php [hunk 5]
```diff
@@ -15,222 +15,224 @@ class EventController extends Controller
...
+                    $option->iscorrect = 0;
+                else $option->iscorrect = 1;
+                $option->save();
+            }
+        }
+        return back()->with('success','Question added successfully.');
+    }
+
+    public function editEvent(Request $request, $id){
+        $event = Event::findOrFail($id);
```
Comment (ankitjain28may) [inline app/Http/Controllers/EventController.php]: Every where you have used findOrFail, its good to use findOrFail but its adviced to use proper error handling by showing the message to the user that this event/whatever is invalid or not found rather showing 404 page for every error

File: app/Http/Controllers/EventController.php [hunk 6]
```diff
@@ -15,222 +15,224 @@ class EventController extends Controller
...
+                $event->duration = $request->duration;
+                $event->correctmark = $request->correct_mark;
+                $event->wrongmark = $request->wrong_mark;
+                $event->quedisplay = $request->display_ques;
+
+                $event->save();
+                return redirect('teacher')->with('edit','Event Edited Successfully');
+    }
+    public function editQue(Request $request, $id, $qid){
+        $editque = Queans::findOrFail($qid);
```
Comment (ankitjain28may) [inline app/Http/Controllers/EventController.php]: here too

**Comment by ankitjain28may** [comment]
Is login through infoconnect working ?

File: app/Http/Controllers/EventController.php (hunk 5 above)
Comment (anantjain6) [reply app/Http/Controllers/EventController.php]: May I design a separate page for all bad request?

**Comment by anantjain6** [comment]
es, its working

**Comment by ankitjain28may** [comment]
@anantjain6 I am unable to login with my infoconnect credentials

File: app/Http/Controllers/EventController.php (hunk 5 above)
Comment (ankitjain28may) [reply app/Http/Controllers/EventController.php]: You can check like this- `$sub = Subject::select('id')->where('id',$request->subject)->first();`
and return back with error message as you did before

**Comment by anantjain6** [comment]
Sir, its because API is connected with dummy DB

**Comment by ankitjain28may** [comment]
Okay, let me know when its working, I want to test

**Comment by anantjain6** [comment]
Ok sir, I'll let you know

**Comment by anantjain6** [comment]
Commit eb24488 added

**Pull request merged**